### PR TITLE
Move VLCMediaFileDiscovererDelegate to VLCMediaLibraryManager

### PR DESCRIPTION
**Pull request by Mikanbu**
### Checklist
- [x] I've run `bundle exec fastlane test` from the root directory to see all new and existing tests pass
- [x] I've followed the [vlc-ios code style](Docs/CodingStyle.md) and run `bundle exec fastlane lint` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://gfrithub.com/videolan/vlc-ios#contribute)
- [x] I've updated the documentation if necessary.

### Description

This allows monitoring of changes in the Documents files from external
apps.

Additionally changes `mediaFilesFoundRequiringAdditionToStorageBackend`
to optional since the new medialibrary doesn't need it.

### CODE REVIEW

File: Sources/VLCAppDelegate.m [hunk 1]
```diff
@@ -283,36 +277,6 @@ - (void)application:(UIApplication *)application performActionForShortcutItem:(U
...
-#pragma mark - media discovering
-
-- (void)mediaFileAdded:(NSString *)fileName loading:(BOOL)isLoading
-{
-    if (!isLoading) {
-        MLMediaLibrary *sharedLibrary = [MLMediaLibrary sharedMediaLibrary];
-        [sharedLibrary addFilePaths:@[fileName]];
-
-        /* exclude media files from backup (QA1719) */
-        NSURL *excludeURL = [NSURL fileURLWithPath:fileName];
```
Comment (carolanitz) [inline Sources/VLCAppDelegate.m]: can you make sure that we exclude mediafiles in the future also from backup?

Comment (Mikanbu) [reply Sources/VLCAppDelegate.m]: By mediafiles do you mean all the users media?

Comment (carolanitz) [reply Sources/VLCAppDelegate.m]: yes

Comment (Mikanbu) [reply Sources/VLCAppDelegate.m]: Updated!

File: SharedSources/VLCMediaLibraryManager.swift [hunk 2]
```diff
@@ -219,6 +228,26 @@ extension VLCMediaLibraryManager {
...
+    func mediaFileAdded(_ filePath: String!, loading isLoading: Bool) {
+        guard !isLoading else {
+            return
+        }
+        /* exclude media files from backup (QA1719) */
+        var excludeURL = URL(fileURLWithPath: filePath)
+        var resourceValue = URLResourceValues()
+
+        resourceValue.isExcludedFromBackup = true
+        try? excludeURL.setResourceValues(resourceValue)
```
Comment (carolanitz) [inline SharedSources/VLCMediaLibraryManager.swift]: can you catch the failure here ?

File: Sources/VLCMediaFileDiscoverer.h [hunk 3]
```diff
@@ -14,10 +14,8 @@
 
 @protocol VLCMediaFileDiscovererDelegate <NSObject>
 
-@required
-- (void)mediaFilesFoundRequiringAdditionToStorageBackend:(NSArray <NSString *> *)foundFiles;
-
 @optional
+- (void)mediaFilesFoundRequiringAdditionToStorageBackend:(NSArray <NSString *> *)foundFiles;
```
Comment (carolanitz) [inline Sources/VLCMediaFileDiscoverer.h]: why did you move it to optional and do we still need it if it's not implemented?

File: SharedSources/VLCMediaLibraryManager.swift [hunk 4]
```diff
@@ -219,6 +228,26 @@ extension VLCMediaLibraryManager {
...
+
+        reload()
+    }
+
+    func mediaFileDeleted(_ filePath: String!) {
+        reload()
```
Comment (carolanitz) [inline SharedSources/VLCMediaLibraryManager.swift]: We should not have an entire reload for mediaFileDeleted. Instead we should just remove from the collectionView the file that belongs to that path

**Comment by carolanitz** [comment]
there are still commented out references in the Appdelegate

**Comment by Mikanbu** [comment]
> there are still commented out references in the Appdelegate

Are you talking about l268? If so, it's unrelated to this PR, it should be addressed in the migration one since it touches the old migration code.

As for removing the use of `VLCMediafileDiscoverer` inside `VLCDocumentPickerController`, I don't think it should be addressed in this PR as well.

File: Sources/VLCMediaFileDiscoverer.h (hunk 3 above)
Comment (Mikanbu) [reply Sources/VLCMediaFileDiscoverer.h]: As I can see it's used in tvOS and can't say if it's really needed but I really don't want to start refactoring tvOS related code here. 

Additionally, my goal for this PR is not the remove the usage of `VLCMediaFileDiscoverer`.

For the why I removed it, as written in the commit description, I changed it to optional since the new medialibrary doesn't need a callback for `mediaFilesFoundRequiringAdditionToStorageBackend`.

File: SharedSources/VLCMediaLibraryManager.swift (hunk 4 above)
Comment (Mikanbu) [reply SharedSources/VLCMediaLibraryManager.swift]: Here, I'm reloading the medialibrary not the UI.
UI related updates(addition/deletion by indexes) should be addressed in another PR in the respective models.

File: SharedSources/VLCMediaLibraryManager.swift (hunk 2 above)
Comment (Mikanbu) [reply SharedSources/VLCMediaLibraryManager.swift]: Will do for the failure catching!

**Comment by carolanitz** [comment]
> Are you talking about l268? If so, it's unrelated to this PR, it should be addressed in the migration one since it touches the old migration code.

yes l268 but there is also still a call in the migration completion. For me it would've made sense since you're moving the entire Mediafilediscoverer but that's fine
